bundles/org.openhab.core.automation.module.script/src/main/java/org/openhab/core/automation/module/script/internal/handler/AbstractScriptModuleHandler.java

```diff
@@ -45,30 +49,28 @@
     /** Constant defining the configuration parameter of modules that specifies the script itself */
     protected static final String SCRIPT = "script";
 
-    protected ScriptEngineManager scriptEngineManager;
+    protected final ScriptEngineManager scriptEngineManager;
 
     private final String engineIdentifier;
 
     private Optional<ScriptEngine> scriptEngine = Optional.empty();
-    private String type;
-    protected String script;
+    private @NonNullByDefault({}) String type;
+    protected @NonNullByDefault({}) String script;
```

Comment:
We introduced the usage of `@NonNullByDefault({})` for variables we know that they are not null but this cannot be detected by the tooling. One common example has been references of components that are injected by the declarative service implementation on activation.
For this use case there is no need anymore because every mandatory reference can now be set by using constructor injection.
For optional references, we should already use nullable because they can be valid null.
Another use case are members that are injected by reflection because of specific deserialization (e.g. Gson).

Why do we need it in this case.
Wouldn't it be correct to mark these variable `@Nullable`?